### ISSUE TITLE
Cannot find module 'react-dom/lib/ReactPortal'

### ISSUE THREAD

**Issue by ForbesLindesay** [manaflair/mylittledom#3]
I keep getting the error:

> Cannot find module 'react-dom/lib/ReactPortal'

when I try to run the react example.

**Comment by ForbesLindesay** [comment]
After updating to react "16.0.0-alpha.6" (and deleting npm's new package-lock.json), that error is now fixed.

I now get:

```
TypeError: this.textLayout.reset(...).apply is not a function
    at TermText.clearTextLayoutCache (<root>i/node_modules/@manaflair/mylittledom/term/elements/TermTextBase.js:508:33)
    at trigger (<root>i/node_modules/@manaflair/mylittledom/term/elements/TermTextBase.js:173:23)
    at TermText.set (<root>i/node_modules/@manaflair/mylittledom/core/dom/Node.js:219:17)
    at TermText.setPropertyTrigger (<root>i/node_modules/@manaflair/mylittledom/core/dom/Node.js:223:20)
    at TermTextBase (<root>i/node_modules/@manaflair/mylittledom/term/elements/TermTextBase.js:134:14)
    at TermText (<root>i/node_modules/@manaflair/mylittledom/term/elements/TermText.js:39:17)
    at createTextInstance (<root>i/node_modules/@manaflair/mylittledom/term/react.js:430:16)
    at completeWork (<root>i/node_modules/react-dom/lib/ReactFiberCompleteWork.js:261:32)
    at completeUnitOfWork (<root>i/node_modules/react-dom/lib/ReactFiberScheduler.js:544:18)
    at performUnitOfWork (<root>i/node_modules/react-dom/lib/ReactFiberScheduler.js:641:14
```

**Comment by ForbesLindesay** [comment]
If I remove the `.apply` stuff, it just renders a blank screen.

**Comment by ForbesLindesay** [comment]
It looks like the `TextOperation` class in text-layout is not properly being wrapped in the JavaScript class

**Comment by ForbesLindesay** [comment]
For reference:

```
npm --versions
{ npm: '5.0.1',
  ares: '1.10.1-DEV',
  cldr: '31.0.1',
  http_parser: '2.7.0',
  icu: '59.1',
  modules: '57',
  node: '8.0.0',
  openssl: '1.0.2k',
  tz: '2017b',
  unicode: '9.0',
  uv: '1.11.0',
  v8: '5.8.283.41',
  zlib: '1.2.11' }
```

and I'm on OSX

**Comment by arcanis** [comment]
> Cannot find module 'react-dom/lib/ReactPortal'

Yep, you have this error because later React releases do not expose these elements anymore. I believe that it [should be fixed before the final release of React 16](https://github.com/facebook/react/issues/9103#issuecomment-304293164), fingers crossed! In the meantime, you'll have to use the alpha.6.

> TypeError: this.textLayout.reset(...).apply is not a function

This error is due to node native modules. For some super-weird reason, if two native modules export the same symbols, they will clash with each other. Unfortunately, both `text-buffer` and `text-layout` export the same `Patch` class, which break nbind (the library I use to compile my native code to both native modules & asm.js) ... I think the latest release should [ship with a fix](https://github.com/charto/nbind/issues/60), but I'm not completely sure it will work.

**Comment by arcanis** [comment]
I found a fix! If you remove your node_modules and reinstall mylittledom it should be fixed (you'll still need react@16.0.0-alpha.6, tho)! :)
